apps/emqx_conf/src/emqx_conf_certs.erl

```diff
@@ -0,0 +1,266 @@
+            ({Node, {Class, Reason}}) ->
+                {true, #{node => Node, kind => rpc, reason => {Class, Reason}}}
+        end,
+        NodeRes
+    ),
+    case Errors of
+        [] ->
+            ok;
+        [_ | _] ->
+            {error, Errors}
+    end.
+
+%%------------------------------------------------------------------------------
+%% RPC Targets
+%%------------------------------------------------------------------------------
+
+-spec add_managed_file_v1(maybe_namespace(), bundle_name(), file_kind(), contents()) ->
+    ok | {error, file:posix()}.
+-doc #{since => <<"6.1.0">>}.
+add_managed_file_v1(Namespace, BundleName, Kind, Contents) ->
+    Filename = filename(Namespace, BundleName, Kind),
+    maybe
+        ok ?= filelib:ensure_dir(Filename),
+        file:write_file(Filename, Contents)
+    end.
+
+-spec delete_managed_file_v1(maybe_namespace(), bundle_name(), file_kind()) ->
+    ok | {error, file:posix()}.
+-doc #{since => <<"6.1.0">>}.
+delete_managed_file_v1(Namespace, BundleName, Kind) ->
+    Filename = filename(Namespace, BundleName, Kind),
+    file:delete(Filename).
+
+-spec delete_bundle_v1(maybe_namespace(), bundle_name()) ->
+    ok | {error, file:posix()}.
+-doc #{since => <<"6.1.0">>}.
+delete_bundle_v1(Namespace, BundleName) ->
+    Dir = dir(Namespace, BundleName),
+    file:del_dir_r(Dir).
+
+%%------------------------------------------------------------------------------
+%% Internal fns
+%%------------------------------------------------------------------------------
+
+filename(Namespace, BundleName, ?FILE_KIND_KEY) ->
+    filename:join(dir(Namespace, BundleName), ?FILENAME_KEY);
+filename(Namespace, BundleName, ?FILE_KIND_CHAIN) ->
+    filename:join(dir(Namespace, BundleName), ?FILENAME_CHAIN);
+filename(Namespace, BundleName, ?FILE_KIND_CA) ->
+    filename:join(dir(Namespace, BundleName), ?FILENAME_CA);
+filename(Namespace, BundleName, ?FILE_KIND_ACC_KEY) ->
+    filename:join(dir(Namespace, BundleName), ?FILENAME_ACC_KEY);
+filename(Namespace, BundleName, ?FILE_KIND_KEY_PASSWORD) ->
+    filename:join(dir(Namespace, BundleName), ?FILENAME_KEY_PASSWORD).
+
+base_dir(?global_ns) ->
+    DataDir = emqx:data_dir(),
+    filename:join([DataDir, certs2, global]);
+base_dir(Namespace0) when is_binary(Namespace0) ->
+    DataDir = emqx:data_dir(),
+    Namespace = escape_name(Namespace0),
+    filename:join([DataDir, certs2, ns, Namespace]).
+
+dir(Namespace, BundleName0) ->
+    BaseDir = base_dir(Namespace),
+    BundleName = escape_name(BundleName0),
+    filename:join([BaseDir, BundleName]).
+
+filename_to_kind(?FILENAME_KEY) ->
+    {ok, ?FILE_KIND_KEY};
+filename_to_kind(?FILENAME_CHAIN) ->
+    {ok, ?FILE_KIND_CHAIN};
+filename_to_kind(?FILENAME_CA) ->
+    {ok, ?FILE_KIND_CA};
+filename_to_kind(?FILENAME_ACC_KEY) ->
+    {ok, ?FILE_KIND_ACC_KEY};
+filename_to_kind(?FILENAME_KEY_PASSWORD) ->
+    {ok, ?FILE_KIND_KEY_PASSWORD};
+filename_to_kind(_) ->
+    error.
+
+escape_name(Name) ->
```

Comment:
maybe this should reside in emqx_mt app e.g.
emqx_mt:escape_ns
to be able to reverse, there should be a emqx_mt:unescape_ns when e.g. to be able to find the namespace from dir name.

Reply:
for bundle name, we can restrict it to a safe char-set.